src/preferences/dialog/dlgprefeffects.cpp

```diff
@@ -12,64 +12,83 @@ DlgPrefEffects::DlgPrefEffects(QWidget* pParent,
+    for (auto profile : m_availableEffectsModel.profiles()) {
+        EffectManifestPointer pManifest = profile->pManifest;
+
+        // Users are likely to have lots of external plugins installed and 
+        // many of them are useless for DJing. To avoid cluttering the list 
+        // shown in WEffectSelector, blacklist external plugins by default.
+        bool defaultValue = (pManifest->backendType() == EffectBackendType::BuiltIn);
+        bool visible = m_pConfig->getValue<bool>(ConfigKey("[Visible Effects]", 
+                                                 pManifest->id()), defaultValue);
+        profile->bIsVisible = visible;
+        m_pEffectsManager->setEffectVisibility(pManifest, visible);
+    }
+    availableEffectsList->setModel(&m_availableEffectsModel);
+
+    connect(availableEffectsList->selectionModel(),
+            SIGNAL(currentRowChanged(const QModelIndex&, const QModelIndex&)),
             this,
-            SLOT(slotEffectSelected(QListWidgetItem*, QListWidgetItem*)));
+            SLOT(availableEffectsListItemSelected(const QModelIndex&)));
+
+    // Highlight first row
+    availableEffectsList->selectRow(0);
+
+  #if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
+    availableEffectsList->horizontalHeader()->setResizeMode(0, QHeaderView::ResizeToContents);
+    availableEffectsList->setColumnWidth(1, 200);
+    availableEffectsList->horizontalHeader()->setResizeMode(2, QHeaderView::ResizeToContents);
+  #else
+    availableEffectsList->horizontalHeader()->setSectionResizeMode(0, QHeaderView::ResizeToContents);
+    availableEffectsList->setColumnWidth(1, 200);
+    availableEffectsList->horizontalHeader()->setSectionResizeMode(2, QHeaderView::ResizeToContents);
+  #endif // QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
+}
+
+DlgPrefEffects::~DlgPrefEffects() {
 }
 
 void DlgPrefEffects::slotUpdate() {
     clear();
-    const QList<EffectManifestPointer> availableEffectManifests =
-            m_pEffectsManager->getAvailableEffectManifests();
-
-    for (const auto& pManifest : availableEffectManifests) {
-        QListWidgetItem* pItem = new QListWidgetItem();
-        pItem->setData(Qt::UserRole, pManifest->id());
-        pItem->setText(pManifest->displayName());
-        availableEffectsList->addItem(pItem);
-    }
+    m_availableEffectsModel.resetFromEffectManager(m_pEffectsManager);
 
-    if (!availableEffectManifests.isEmpty()) {
-        availableEffectsList->setCurrentRow(0);
+    if (!m_availableEffectsModel.isEmpty()) {
+        availableEffectsList->selectRow(0);
     }
 }
 
 void DlgPrefEffects::slotApply() {
-    // Nothing to apply.
+    for (EffectProfilePtr profile : m_availableEffectsModel.profiles()) {
+        EffectManifestPointer pManifest = profile->pManifest;
+        m_pEffectsManager->setEffectVisibility(pManifest, profile->bIsVisible);
+        m_pConfig->set(ConfigKey("[Visible Effects]", pManifest->id()), ConfigValue(profile->bIsVisible));
```

Comment:
Now that we have noticed the potential for name collisions using just the ID string of EffectManifests, I see that problem could also occur here. I propose adding the EffectBackendType converted to a string to the group of the ConfigKey, for example "[Visible Built-in Effects]". We'll need a new public method of EffectManifest, let's call it backendTypeFromUntranslatedString, so the logic works regardless of the user's locale.